app/controllers/beaches_controller.rb

```diff
@@ -65,6 +65,41 @@ def destroy
 
   private
 
+  def search(query)
+    if query == nil
+      @beaches = Beach.all
+    elsif query.match(/^\s*$/)
+      flash[:notice] = "Please enter one or more words to search."
+      @beaches = Beach.all
+    elsif query.split.size == 1
+      @beaches = Beach.joins(:amenities).where(
+        "beaches.name ILIKE ? OR
+        beaches.description ILIKE ? OR
+        beaches.city ILIKE ? OR
+        beaches.state ILIKE ? OR
+        beaches.zip ILIKE ? OR
+        amenities.name ILIKE ?",
+        "%#{query}%", "%#{query}%", "%#{query}%", "%#{query}%", "%#{query}%",
+        "%#{query}%"
+      )
+    elsif query.split.size >= 2
+      container = []
+      query_string = query.split.map do
+        "(beaches.name ILIKE ? OR
+        beaches.description ILIKE ? OR
+        beaches.city ILIKE ? OR
+        beaches.state ILIKE ? OR
+        beaches.zip ILIKE ? OR
+        amenities.name ILIKE ?) AND "
+      end
+
+      container << query_string.inject(:+).rpartition(" AND ").first
+      query.split.each { |word| 6.times { container << "%#{word}%" } }
+
+      @beaches = Beach.joins(:amenities).where(container.flatten)
+    end
+  end
```

Comment:
Per my comment above, this should be a class method in the `Beach` model that takes a query as an argument. Rewrite that accordingly (some of the logic in here will go away, for example).